core/src/main/scala/akka/kafka/javadsl/Producer.scala

```diff
@@ -192,6 +194,32 @@ object Producer {
       .asJava
       .asInstanceOf[Flow[Envelope[K, V, PassThrough], Results[K, V, PassThrough], NotUsed]]
 
+  /**
+   * API MAY CHANGE
+   *
+   * Create a flow to conditionally publish records to Kafka topics and then pass it on.
+   *
+   * It publishes records to Kafka topics conditionally:
+   *
+   * - [[akka.kafka.ProducerMessage.Message Message]] publishes a single message to its topic, and continues in the stream as [[akka.kafka.ProducerMessage.Result Result]]
+   *
+   * - [[akka.kafka.ProducerMessage.MultiMessage MultiMessage]] publishes all messages in its `records` field, and continues in the stream as [[akka.kafka.ProducerMessage.MultiResult MultiResult]]
+   *
+   * - [[akka.kafka.ProducerMessage.PassThroughMessage PassThroughMessage]] does not publish anything, and continues in the stream as [[akka.kafka.ProducerMessage.PassThroughResult PassThroughResult]]
+   *
+   * This flow is intended to be used with Akka's [flow with context](https://doc.akka.io/docs/akka/current/stream/operators/Flow/asFlowWithContext.html).
+   */
+  @ApiMayChange
+  def flowWithContext[K, V, C](
+      settings: ProducerSettings[K, V]
+  ): Flow[Pair[Envelope[K, V, NotUsed], C], Pair[Results[K, V, C], C], NotUsed] =
```

Comment:
When I tried to use `javadsl.FlowWithContext` as return type instead it didn't compose in `via`. I still need to understand what I did wrong.

Reply:
That is a bug https://github.com/akka/akka/issues/26896

Reply:
Thank you for having a look, I sensed it might not be me this time.